src/app/core/entity-components/entity-utils/dynamic-form-components/edit-single-entity/edit-single-entity.component.html

```diff
@@ -1,23 +1,37 @@
-<mat-form-field [formGroup]="formControl.parent">
+<mat-form-field *ngIf="!selectedEntity || editingSelectedEntity" [formGroup]="formControl.parent">
   <mat-label>{{ label }}</mat-label>
   <input
-    #inputField
+    #inputElement
     matInput
     [fxHide]="!formControl.enabled"
     [placeholder]="placeholder"
     [formControl]="entityNameFormControl"
     [matAutocomplete]="autoSuggestions"
+    (keyup.enter)="submit()"
+    (focusout)="submit()"
   />
   <mat-autocomplete
     #autoSuggestions="matAutocomplete"
+    (optionSelected)="select($event)"
```

Comment:
Good fix! Currently selecting a option with the 'enter' button results in an error because `formControl.setValue` is never called. This fixes the problem.